app/Libraries/BeatmapsetDiscussionPostHandlesProblem.php

```diff
@@ -0,0 +1,60 @@
+<?php
+
+// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the GNU Affero General Public License v3.0.
+// See the LICENCE file in the repository root for full licence text.
+
+declare(strict_types=1);
+
+namespace App\Libraries;
+
+use App\Jobs\Notifications;
+use App\Models\BeatmapDiscussion;
+use App\Models\User;
+
+abstract class BeatmapsetDiscussionPostHandlesProblem
+{
+    protected int $priorOpenProblemCount = 0;
```

Comment:
this probably doesn't need to be a count but a boolean of `exists()` check instead